app_legacy/Platform/Commands/UpdatePlayerMasteries.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LegacyApp\Platform\Commands;
+
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\DB;
+use LegacyApp\Community\Enums\AwardType;
+use LegacyApp\Platform\Enums\AchievementType;
+use LegacyApp\Platform\Enums\UnlockMode;
+use LegacyApp\Site\Models\User;
+
+class UpdatePlayerMasteries extends Command
+{
+    protected $signature = 'ra-legacy:platform:update-player-masteries {username?}';
+
+    protected $description = 'Recalculates masteries for a user';
+
+    private $gameAchievements = [];
+
+    public function handle(): void
+    {
+        $username = $this->argument('username');
+        if (!empty($username)) {
+            $this->recalculate($username);
+
+            return;
+        }
+
+        $users = DB::connection('mysql_legacy')
+            ->table('SiteAwards')
+            ->where('AwardType', '=', AwardType::Mastery)
```

Comment:
`DB::connection('mysql_legacy')>table('SiteAwards')` can usually be replaced by using the `PlayerBadge` model's facade directly (which is bound to the `SiteAwards` table), e.g.:

```php
use LegacyApp\Platform\Models\PlayerBadge;

PlayerBadge::where('AwardType', '=', AwardType::Mastery)
    ->where('AwardType', '=', AwardType::Mastery)
    ->distinct()
    ->get(['User'])
```

I will however touch that in the mysql refactoring PR - so no need to do that right now.